src/core.js

```diff
@@ -850,9 +852,43 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
     }
   }
 
+  /**
+   * Get instance of index mapper which is responsible for managing the row indexes.
+   *
+   * @return {IndexMapper}
+   */
+  this.getRowIndexMapper = function() {
+    return this.rowIndexMapper;
+  };
+
+  /**
+   * Get instance of index mapper which is responsible for managing the column indexes.
+   *
+   * @return {IndexMapper}
+   */
+  this.getColumnIndexMapper = function() {
+    return this.columnIndexMapper;
+  };
+
+  /**
+   * Execute batch operations on Handsontable instance with updating cache.
+   *
+   * @param {Function} curriedBatchOperations Batched operations curried in a function.
+   */
+  this.executeBatchOperations = function(curriedBatchOperations) {
+    this.getColumnIndexMapper().executeBatchOperations(() => {
+      this.getRowIndexMapper().executeBatchOperations(() => {
+        curriedBatchOperations();
+      });
+    });
+  };
+
   this.init = function() {
     dataSource.setData(priv.settings.data);
-    instance.runHooks('beforeInit');
+
+    instance.executeBatchOperations(() => {
+      instance.runHooks('beforeInit');
+    });
```

Comment:
There is already one unit tests for mentioned function: https://github.com/handsontable/handsontable/blob/2a82267bb7d4033849ef09edc50a208a01be1c68/test/unit/translations/indexMapper.spec.js#L965-L990

However, there is no integration test for that. Do you think that it's really necessary? It's related to the fact that `beforeInit` enable all plugins: https://github.com/handsontable/handsontable/blob/55aa59de30669a208815aa3a3d64c255681978ce/src/plugins/_base.js#L36

When we enabling plugin, there are some operations, such as `registerMap` which may reset the cache. 

https://github.com/handsontable/handsontable/blob/55aa59de30669a208815aa3a3d64c255681978ce/src/plugins/trimRows/trimRows.js#L76

Maybe just extra comment describing why we use it there will be enough? WDYT?

Reply:
@wszymanski I was going to suggest a test for `beforeInit` to ensure that `updateCache` is called only once even if multiple plugins did some changes in `indexMapper`.

@warpech Looks random because it was an optimization for plugin init only. YAGNI? ;) And I see some risks if we wrap all `runHooks` calls in `executeBatchOperations`. I think we should discuss both issues with `executeBatchOperations` that were mentioned by you with the team, once everyone finishes with their reviews and we're on the same page.

Reply:
Do we have the same for plugin `destroy` calls? There's no point in rebuilding cache until we disable them all. And we should check `updatePlugin` to avoid cache rebuild on `disable` and then `enablePlugin`.